src/ros2idl.ne

```diff
@@ -0,0 +1,395 @@
+@{%
+
+// necessary to use keywords to avoid using the `reject` postprocessor which can cause poor perf
+// having these as keywords removes ambiguity with `customType` rule
+const keywords = [
+  , "struct"
+  , "module"
+  , "const"
+  , "include"
+
+  //types
+  , "boolean"
+  , "wstring"
+  , "string"
+  , "sequence"
+	
+  // numeric types
+  , "byte"
+  , "octet"
+  , "wchar"
+  , "char"
+  , "double"
+  , "float"
+  , "int8"
+  , "uint8"
+  , "int16"
+  , "uint16"
+  , "int32"
+  , "uint32"
+  , "int64"
+  , "uint64"
+  , "unsigned" 
+  , "short"  
+  , "long"
+];
+
+const kwObject = keywords.reduce((obj, w) => {
+	obj[w] = w;
+	return obj;
+}, {});
+
+const moo = require("moo");
+// Terminal tokens are in all caps
+const lexer = moo.compile({
+  SPACE: {match: /\s+/, lineBreaks: true},
+  DECIMALEXP: /(?:(?:\d+\.\d*)|(?:\d*\.\d+)|(?:[0-9]+))[eE](?:[+|-])?[0-9]+/,
+  DECIMAL: /(?:(?:\d+\.\d*)|(?:\d*\.\d+))/,
+  INTEGER: /[0-9]+/,
+  COMMENT: /(?:\/\/[^\n]*)|(?:\/\*(?:.|\n)+?\*\/)/,
+  HEX_LITERAL: /0x(?:[0-9][a-f][A-F])+?/,
+  STRING: [{match: /"(?:\\["\\rnu]|[^"\\])*?"/, lineBreaks: true, value: x => x.slice(1, -1)}], // remove outside quotes
+  LCBR: '{',
+  RCBR: '}',
+  LBR: '[',
+  RBR: ']',
+  LT: '<',
+  GT: '>',
+  LPAR: '(',
+  RPAR: ')',
+  ';': ';',
+  ',': ',',
+  AT: '@',
+  PND: '#',
+  PT: ".",
+  '/': "/",
+  SIGN: /[+|-]/,
+  EQ: /=[^\n]*?/,
+  NAME: {match: /[a-zA-Z_][a-zA-Z0-9_]*(?:\:\:[a-zA-Z][a-zA-Z0-9_]*)*/, type: moo.keywords(kwObject)},
+});
+
+// Utiility functions
+
+// also used to parse tokens to strings since they start as an object
+function join(d){
+	return d.join("");
+}
+
+// used for combining AST components
+function extend(objs) {
+  return objs.reduce((r, p) => ({ ...r, ...p }), {});
+}
+
+function noop() {
+	return null;
```

Comment:
unfortunately I'm not seeing another way to run prettier on this without making a prettier plugin

Reply:
Depending on how you want to organize things - nearly does provide a way to use external js files: 
<img width="815" alt="image" src="https://user-images.githubusercontent.com/84792/222476153-d9496877-e2ca-4f71-9068-7cf8b0395f60.png">

Not sure if that ends up being better but it might solve some of your formatting woes